presto-main/src/main/java/com/facebook/presto/operator/aggregation/BlackholeAggregation.java

```diff
@@ -0,0 +1,137 @@
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.metadata.SqlAggregationFunction;
+import com.facebook.presto.operator.aggregation.state.NullableLongState;
+import com.facebook.presto.operator.aggregation.state.StateCompiler;
+import com.facebook.presto.spi.function.aggregation.Accumulator;
+import com.facebook.presto.spi.function.aggregation.AggregationMetadata;
+import com.facebook.presto.spi.function.aggregation.AggregationMetadata.AccumulatorStateDescriptor;
+import com.facebook.presto.spi.function.aggregation.GroupedAccumulator;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.common.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.spi.function.Signature.comparableTypeParameter;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INDEX;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.NULLABLE_BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.slice.Slices.utf8Slice;
+
+/**
+ * Aggregating function that does nothing on the input data and always returns
+ * the "OK" varchar value.
+ *
+ * It works similar to the CHECKSUM function, but it does not compute the hash to
+ * minimize resource usage. It can be used when you need to scan all data in a table.
+ */
+public class BlackholeAggregation
+        extends SqlAggregationFunction
+{
+    public static final BlackholeAggregation BLACKHOLE_AGGREGATION = new BlackholeAggregation();
+    private static final String NAME = "blackhole";
+    private static final Slice OUTPUT = utf8Slice("OK");
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(BlackholeAggregation.class, "output", NullableLongState.class, BlockBuilder.class);
+    private static final MethodHandle INPUT_FUNCTION = methodHandle(BlackholeAggregation.class, "input", Type.class, NullableLongState.class, Block.class, int.class);
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(BlackholeAggregation.class, "combine", NullableLongState.class, NullableLongState.class);
+
+    public BlackholeAggregation()
+    {
+        super(NAME,
+                ImmutableList.of(comparableTypeParameter("T")),
```

Comment:
i don't think you care if the input type is comparable.  Can instead call `typeVariable("T")`